ms/MSOper/MSMetaData.cc

```diff
@@ -252,7 +254,34 @@ uInt MSMetaData::nRows(CorrelationType cType) {
 	}
 }
 
-uInt MSMetaData::nRows(
+SHARED_PTR<const map<SubScanKey, uInt64> > MSMetaData::getNRowMap(CorrelationType cType) const {
+    uInt64 nACRows, nXCRows;
+    SHARED_PTR<std::map<SubScanKey, uInt64> > subScanToNACRowsMap, subScanToNXCRowsMap;
+    SHARED_PTR<vector<uInt64> > fieldToNACRowsMap, fieldToNXCRowsMap;
+    _getRowStats(
+        nACRows, nXCRows, subScanToNACRowsMap,
+        subScanToNXCRowsMap, fieldToNACRowsMap,
+        fieldToNXCRowsMap
+    );
+    if (cType == AUTO) {
+        return subScanToNACRowsMap;
+    }
+    else if (cType == CROSS) {
+        return subScanToNXCRowsMap;
+    }
+    SHARED_PTR<map<SubScanKey, uInt64> > mymap(
+        new map<SubScanKey, uInt64>()
+    );
```

Comment:
I guess we don't have a `make_shared` in our abstraction? would save a (probably irrelevant) memory allocation.